dipy/tracking/vox2track.pyx

```diff
@@ -162,6 +161,172 @@ def streamline_mapping(streamlines, voxel_size=None, affine=None,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+cdef inline double norm(double x, double y, double z) nogil:
+    cdef double val = sqrt(x*x + y*y + z*z)
+    return val
+
+
+# Changing this to a memview was slower.
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef inline void c_get_closest_edge(double p_x, double p_y, double p_z,
+                                    double d_x, double d_y, double d_z,
+                                    cnp.double_t[:] edge,
+                                    double eps=1.) nogil:
+     edge[0] = floor(p_x + eps) if d_x >= 0.0 else ceil(p_x - eps)
+     edge[1] = floor(p_y + eps) if d_y >= 0.0 else ceil(p_y - eps)
+     edge[2] = floor(p_z + eps) if d_z >= 0.0 else ceil(p_z - eps)
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def streamlines_in_mask(list streamlines,
```

Comment:
Would you say this function should never be called directly, going through dipy.tracking.utils.target_line_based instead? Because if you consider it could be called directly, what you expect for the ```lin``` parameter is not in line with what most dipy functions expect (i.e. the basic affine transform)